test/unit/dialects/mssql/query-generator.test.js

```diff
@@ -0,0 +1,49 @@
+
+'use strict';
+
+/* jshint -W110 */
+var chai = require('chai')
+  , expect = chai.expect
+  , Support = require(__dirname + '/../../support')
+  , dialect = Support.getTestDialect()
+  , _ = require('lodash')
+  , QueryGenerator = require('../../../../lib/dialects/mssql/query-generator');
+
+if (dialect === 'mssql') {
+  describe('[MSSQL Specific] QueryGenerator', function() {
+    var suites = {
+      getDefaultConstraintQuery: [
+        {
+          arguments: [{tableName:'myTable', schema: 'mySchema'}, 'myColumn'],
+          expectation: "SELECT name FROM SYS.DEFAULT_CONSTRAINTS WHERE PARENT_OBJECT_ID = OBJECT_ID('[mySchema].[myTable]', 'U') AND PARENT_COLUMN_ID = (SELECT column_id FROM sys.columns WHERE NAME = ('myColumn') AND object_id = OBJECT_ID('[mySchema].[myTable]', 'U'));"
+        }
+      ],
+      dropConstraintQuery: [
+        {
+          arguments: [{tableName:'myTable', schema: 'mySchema'}, 'myConstraint'],
+          expectation: "ALTER TABLE [mySchema].[myTable] DROP CONSTRAINT [myConstraint];"
+        }
+      ]
+    };
+
+    _.each(suites, function(tests, suiteTitle) {
+      describe(suiteTitle, function() {
+        tests.forEach(function(test) {
+          var title = test.title || 'MSSQL correctly returns ' + test.expectation + ' for ' + JSON.stringify(test.arguments);
+          it(title, function() {
+            // Options would normally be set by the query interface that instantiates the query-generator, but here we specify it explicitly
+            var context = test.context || {options: {}};
+            if (test.needsSequelize) {
+              if (_.isFunction(test.arguments[1])) test.arguments[1] = test.arguments[1](this.sequelize);
+              if (_.isFunction(test.arguments[2])) test.arguments[2] = test.arguments[2](this.sequelize);
+            }
+            QueryGenerator.options = _.assign(context.options, { timezone: '+00:00' });
+            QueryGenerator._dialect = this.sequelize.dialect;
+            var conditions = QueryGenerator[suiteTitle].apply(QueryGenerator, test.arguments);
+            expect(conditions).to.deep.equal(test.expectation);
+          });
+        });
+      });
+    });
+  });
+}
```

Comment:
Mind if I put this in a separate PR? This is consistent with the other pre-existing query-generator tests, and I'd like to avoid making this PR much bigger :)